### DISCUSSION TITLE
pynecone integration

### DISCUSSION THREAD

**Discussion by rraammiinn** [r0x0r/pywebview#1169] [Ideas]
is it possible to combine pywebview with pynecone ?
pynecone is used for creating web apps but I think it's more suitable for desktop and mobile apps duo to it's latency .

**Comment by r0x0r** [comment]
This is the first time I hear about pynecone. 
After glancing over the documentation, integration with _pywebview_ looks possible. Look at the **Self hosting** section in the pynecone documentation for hosting details. You then can point _pywebview_ to the address where pynecone is running. Multiprocessing for _pywebview_ and pynecone might be a good idea

**Reply by rraammiinn** [reply]
but there are two problems .

1.pynecone's initialization time is quite high (because in term of web server it doesn't matter).

2.there is no way to know when pynecone server is ready to launch pywebview windows  (of course you can use sleep but it's a bad idea and an event driven approach seems better) .

**Reply by r0x0r** [reply]
As I see, both issues are with pynecone.

Regarding number 2, if pynecone provides a callback that would be called when pynecone is up and running, you could use that. Otherwise you can periodically check whether server is running and sleep until the next check.

**Reply by rraammiinn** [reply]
thanks .
